### PR TITLE
error handling

### PR DESCRIPTION
**Pull request by jcvenegas**
This PR adds error handling to some functions:

create: Before open console check if is not NULL
log: Add error handling in log setup for hypervisor
process: close duplicated shim fds explicitly
process: handle errors errors in cc_oci_setup_child

### CODE REVIEW

**Comment by chavafg** [comment]
qa-failed

File: tests/util_test.c [hunk 1]
```diff
@@ -517,6 +518,30 @@ START_TEST(test_cc_oci_enable_networking) {
...
+	/* fd pointer is NULL */
+	ck_assert(! dup_over_stdio(NULL));
+
+	/* fd number is -1 */
+	fd = -1;
+	ck_assert(! dup_over_stdio(&fd));
+
+
+	/* dup not used but open STDIN fd */
+	fd = STDIN_FILENO ;
```
Comment (jodh-intel) [inline tests/util_test.c]: This test is going to close whatever `STDIN_FILENO` is attached to. It should probably do something like the following to minimise surprises in the future if the test gets expanded [*]:

```
/* save */
int saved_stdin = dup(STDIN_FILENO).

<< run tests >>

/* restore */
dup2(STDIN_FILENO, saved_stdin);
```

(although of course if an assertion fires that won't help ;)

---

[*] - It should be ok to close stdin in a single `START_TEST/END_TEST` block as they are run in sub-processes.

Comment (jcvenegas) [inline tests/util_test.c]: Good catch! , fixed

**Comment by jcvenegas** [comment]
@jodh-intel thanks for the feedback, now is  fixed, does it look better ? @sameo pullaprove complains about a missing "Signed-off-by" but   I don see anything  missing

**Comment by jodh-intel** [comment]
lgtm.

btw - that pullapprove failure is caused by merge commits that don't have sign-offs (0ae117c, 8eaaf07, 2e70364).

**Comment by jodh-intel** [comment]
(we might need to disable pullapprove to allow this PR through as a result?)

**Comment by grahamwhaley** [comment]
@jodh-intel how did we get merge commits in here - should we not be avoiding merge commits that end up in PRs using rebases?

**Comment by jcvenegas** [comment]
@jodh-intel yes, seems that will need to disable pullaprove, just wonder why other PR does not complain, I'll try to re-rebase master

**Comment by jodh-intel** [comment]
@grahamwhaley - Good question. It looks we might want to ask @dlespiau and @sboeuf. It seems to be easily done, so we need to find a way to disable them going forward to keep the pullapprove gods happy.

File: src/process.c [hunk 2]
```diff
@@ -149,12 +150,19 @@ cc_oci_setup_child (struct cc_oci_config *config)
 
 	/* Do not close fds when VM runs in detached mode*/
 	if (! config->detached_mode) {
-		cc_oci_close_fds (NULL);
+		if (! cc_oci_close_fds (NULL)) {
+			goto out;
```
Comment (devimc) [inline src/process.c]: ```return false``` would be more clear

File: src/process.c [hunk 3]
```diff
@@ -149,12 +150,19 @@ cc_oci_setup_child (struct cc_oci_config *config)
...
+		}
 	}
 
-	cc_oci_setup_hypervisor_logs(config);
+	if (! cc_oci_setup_hypervisor_logs(config)) {
+		goto out;
```
Comment (devimc) [inline src/process.c]: same here

File: src/process.c [hunk 4]
```diff
@@ -149,12 +150,19 @@ cc_oci_setup_child (struct cc_oci_config *config)
...
+	}
 
-	return true;
+	ret = true;
+
+out:
+	return ret;
```
Comment (devimc) [inline src/process.c]: and here just ```return true```

File: src/util.c [hunk 5]
```diff
@@ -606,6 +606,86 @@ cc_oci_handle_signals (void)
...
+		goto close_tmp_fds;
+	} else  {
+		/* success */
+		g_debug("fd moved from %d to  %d", *fdp, dup_fd);
+		*fdp = dup_fd;
+		dup_fd = -1;
+		ret = true;
+	}
+
+close_tmp_fds:
```
Comment (devimc) [inline src/util.c]: seems like this label is not necessary

File: src/util.c [hunk 6]
```diff
@@ -606,6 +606,86 @@ cc_oci_handle_signals (void)
...
+	for (int i = 0 ; i < CC_OCI_ARRAY_SIZE(tmp_fds) ; i++) {
+		/* Dont close the original fd on error */
+		if( ret == false && tmp_fds[i] == *fdp)
```
Comment (devimc) [inline src/util.c]: ``` ! ret ``` looks better

**Comment by sameo** [comment]
LGTM

**Pull request merged**